projects/kit/components/routable-dialog/routable-dialog.component.ts

```diff
@@ -0,0 +1,50 @@
+import {ChangeDetectionStrategy, Component, Inject, Injector, Self} from '@angular/core';
+import {ActivatedRoute, Router, UrlSegment} from '@angular/router';
+import {TuiDestroyService} from '@taiga-ui/cdk';
+import {TuiDialogService} from '@taiga-ui/core';
+import {PolymorpheusComponent} from '@tinkoff/ng-polymorpheus';
+import {takeUntil} from 'rxjs/operators';
+
+@Component({
+    selector: 'tui-routable-dialog',
+    template: '',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [TuiDestroyService],
+})
+export class TuiRoutableDialogComponent {
+    constructor(
+        @Inject(TuiDialogService) private readonly dialogService: TuiDialogService,
```

Comment:
Looks like `dialogService`, `injector` and `destroy$` can be enclosed in constructor without `private readonly`.

Reply:
I think in this case better use `private readonly`. Because these fields are not for public api

Reply:
I mean if you drop `private readonly` completely, they will be enclosed in constructor and not accessible even as private API — you only need them in constructor.

Reply:
Got it, thanks! Refactored